mailpoet/lib/Automation/Integrations/MailPoet/Analytics/RegisterAnalytics.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Analytics;
+
+use MailPoet\API\REST\API;
+use MailPoet\Automation\Engine\Hooks;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\MailPoet\Analytics\Endpoints\OverviewEndpoint;
+
+class RegisterAnalytics {
+
+  /** @var WordPress */
+  private $wordPress;
+
+  public function __construct(
+    WordPress $wordPress
+  ) {
+    $this->wordPress = $wordPress;
+  }
+
+  public function register(): void {
+    $this->wordPress->addAction(Hooks::API_INITIALIZE, function (API $api) {
+      $api->registerPostRoute('automation/analytics/overview', OverviewEndpoint::class);
```

Comment:
Using `POST` doesn't fit very well with the REST API. We're only fetching data, so `GET` would get the correct semantics would (not just on paper, but also regarding how browsers treat caching, etc.). I don't know if WP suggests keeping the RESTful semantics, but I think in this case we can.

Reply:
Thanks for catching! Forgot to change that. `GET` does not allow for a request body somehow, so I skipped that and forgot to change it later. 6dc044548 makes the endpoint now `GET`.